definitions/features/downstream.rb

```diff
@@ -30,6 +30,16 @@ def setup_repositories(version)
     end
   end
 
+  def absent_repos(version)
+    execute!(%(subscription-manager refresh))
```

Comment:
I this is not necessary, sub-man gets a fresh list when you run `subscription-manager repos --list`.

https://github.com/candlepin/subscription-manager/blob/master/src/subscription_manager/managercli.py#L2040

Reply:
I don't see the `repos --list` calling this code: https://github.com/candlepin/subscription-manager/blob/master/src/subscription_manager/managercli.py#L632

Calling the `refresh` was actually suggested by our CEE, when working on the check, and I would prefer keeping it there to stay on safe side here.

Reply:
- This sets `repos` https://github.com/candlepin/subscription-manager/blob/master/src/subscription_manager/managercli.py#L2026
- which calls `RepoUpdateActionCommand` https://github.com/candlepin/subscription-manager/blob/master/src/subscription_manager/repolib.py#L202
- That class gets a refresh list from the server. https://github.com/candlepin/subscription-manager/blob/8b2c0c60e7694694ca78ce7f9b8596f314871f57/src/subscription_manager/repolib.py#L326

Even if you don't follow the code, try enabling a repo on the server for a client, and then running `sub-man repo --list` without doing a `sub-man refresh` and you'll see your newly enabled repo.

Reply:
Still is doesn't trigger the refreshing the entitlements on the server side that way the `rhsm refresh` does https://github.com/candlepin/subscription-manager/blob/master/src/subscription_manager/managercli.py#L629.

The point is that there is an evidence from the field that calling the `rhsm refresh` call helped and I don't want to remove it just because in some cases, it seems the `rhsm repos` can do fine on it's own. And there is no indication that I could see that would mean that `repos --list` is calling exactly the same code as the the `refresh`: it seems that the `refresh` actually is doing something else and we sometimes need it.

What I can do is checking the `rhsm repos` first, and in case the repos is not there, calling `rhsm refresh` and trying agian. However, I would probabely prefer the way it's written right now.

Reply:
`sub-man refresh` I think like you say does do more than `sub-man repos --list`, my point is that `sub-man --list` will always do enough for our use case (get an up-to-date list of enabled repos).

My initial comment was due to the slowness of `sub-man refresh`, our installer and checks are IMO already quite slow and this only makes it worse, but leave it in if you want.

Reply:
And if there's an edge case where `sub-man repos --list` doesn't get an up-to-date list of enabled repos, then that is a `sub-man` bug, we shouldn't need to workaround a potential bug here.

Reply:
Ok, what I did was making the `rhsm refresh` conditional in case the repos don't show up after `rhsm list`, I've also removed the call for `rhsm refresh` that we called later (that was done as fix for https://bugzilla.redhat.com/show_bug.cgi?id=1548368), and after this we don't need to do it twice.

In case the `rhsm list` is enough, we will not call the `refresh` at all, and in case it would not help, I still think it's better to call the `rhsm refresh`, even if that would be the bug on rhsm side, as the users for foreman-maintain don't care that much about where the bug is coming from, but if there is a way on our side to fix the issue for them automatically, we should at least try :)